metrics/lib/common.bash

```diff
@@ -95,18 +95,40 @@ check_images()
 # This function performs a docker build on the image names
 # passed in, to ensure that we have the latest changes from
 # the dockerfiles
+build_dockerfile_image()
+{
+	local image="$1"
+	local dockerfile_path="$2"
+
+	echo "docker building $image"
+	if ! docker build --label "$image" --tag "${image}" -< "$dockerfile_path"; then
+		die "Failed to docker build image $image"
+	fi
+}
+
+# This function verifies that the dockerfile version is
+# equal to the test version in order to build the image or
+# just run the test
 check_dockerfiles_images()
 {
 	local image="$1"
 	local dockerfile_path="$2"
+
 	if [ -z "$image" ] || [ -z "$dockerfile_path" ]; then
 		die "Missing image or dockerfile path variable"
-		exit 1;
 	fi
-	echo "docker building $image"
-	if ! docker build --label "$image" --tag "${image}:latest" -< "$dockerfile_path"; then
-		die "Failed to docker build image $image"
-		exit 1;
+
+	# Verify that dockerfile version is equal to test version
+	check_image=$(docker images "$image" -q)
+	if [ -n "$check_image" ]; then
+		check_image_version=$(docker image inspect $image | grep -w DOCKERFILE_VERSION | head -1 | cut -d '"' -f4)
+		dockerfile_version=$(grep DOCKERFILE_VERSION $dockerfile_path | cut -d '"' -f2)
+		if [ "$dockerfile_version" -ne "$check_image_version" ]; then
```

Comment:
I ran with a `bash -x ./storage/blogbench.sh`, and found this in the output:

```
+ dockerfile_version=1.0
+ '[' 1.0 -ne '' ']'
/home/gwhaley/gopath/src/github.com/kata-containers/tests/metrics/storage/../lib/common.bash: line 126: [: 1.0: integer expression expected
+ metrics_json_init
```

Can you check that for me - it might be because I probably have an image in my docker already without the `DOCKERFILE_VERSION` in it - maybe we can then improve the script capture/handling of that?

Reply:
`-ne` only works with integer comparison so we should either use `!=` or use `bc(1)` or something cleverer still if we have to handle version numbers containing periods.